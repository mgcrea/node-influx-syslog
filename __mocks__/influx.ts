// @NOTE https://jestjs.io/docs/en/manual-mocks.html

const d = (obj: any) => console.dir(obj, {depth: 10, colors: true});

const influx = jest.genMockFromModule('influx');

const __mockedWritePoints = jest.fn().mockImplementation(() => {
  d('writePoints');
});

export class InfluxDB {
  addSchema() {}
  async writePoints() {
    __mockedWritePoints();
  }
}

influx.InfluxDB = InfluxDB;
influx.__mockedWritePoints = __mockedWritePoints;

module.exports = influx;
