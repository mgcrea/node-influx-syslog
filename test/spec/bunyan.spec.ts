import Logger from 'bunyan';
import {
  castSyslogInputPoint,
  BunyanLogObject,
  getSyslogSeverityLevel,
  createSyslogStream
} from '../../src/adapters/bunyan';
import {InfluxDB} from 'influx';

describe('bunyan', () => {
  describe('getSyslogSeverityLevel', () => {
    it('should properly export a class', () => {
      expect(getSyslogSeverityLevel).toBeDefined();
      expect(typeof getSyslogSeverityLevel).toBe('function');
    });
    it('should properly get a syslog levels', () => {
      expect([
        getSyslogSeverityLevel(Logger.TRACE),
        getSyslogSeverityLevel(Logger.DEBUG),
        getSyslogSeverityLevel(Logger.INFO),
        getSyslogSeverityLevel(Logger.WARN),
        getSyslogSeverityLevel(Logger.ERROR),
        getSyslogSeverityLevel(Logger.FATAL)
      ]).toMatchSnapshot();
    });
  });
  describe('castSyslogInputPoint', () => {
    it('should properly export a class', () => {
      expect(castSyslogInputPoint).toBeDefined();
      expect(typeof castSyslogInputPoint).toBe('function');
    });
    it('should properly cast a syslog point', () => {
      const logObject: BunyanLogObject = {
        time: new Date(1562361784183000000),
        level: Logger.DEBUG,
        name: 'fooname',
        facility: 'foofacility',
        pid: '1',
        msg: 'bazqux',
        hostname: 'hostname'
      };
      expect(castSyslogInputPoint(logObject)).toMatchSnapshot();
    });
  });
  describe('createSyslogStream', () => {
    const influx = new InfluxDB('http://localhost:8086/syslog');
    it('should properly export a class', () => {
      expect(createSyslogStream).toBeDefined();
      expect(typeof createSyslogStream).toBe('function');
    });
    it('should properly create a stream', () => {
      const writableStream = createSyslogStream(influx);
      expect(typeof writableStream).toBe('object');
      expect(writableStream).toMatchSnapshot();
    });
  });
});
