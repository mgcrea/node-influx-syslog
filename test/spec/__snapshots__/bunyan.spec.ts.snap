// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bunyan castSyslogInputPoint should properly cast a syslog point 1`] = `
Object {
  "fields": Object {
    "facility_code": 14,
    "message": "bazqux",
    "procid": "1",
    "severity_code": 7,
    "timestamp": "NaN000000",
    "version": 1,
  },
  "measurement": "syslog",
  "tags": Object {
    "appname": "app",
    "facility": "foofacility",
    "host": "hostname",
    "hostname": "hostname",
    "severity": "debug",
  },
}
`;

exports[`bunyan createSyslogStream should properly create a stream 1`] = `
DestroyableTransform {
  "_destroyed": false,
  "_events": Object {
    "prefinish": [Function],
  },
  "_eventsCount": 1,
  "_maxListeners": undefined,
  "_readableState": ReadableState {
    "awaitDrain": 0,
    "buffer": BufferList {
      "head": null,
      "length": 0,
      "tail": null,
    },
    "decoder": null,
    "defaultEncoding": "utf8",
    "destroyed": false,
    "emitClose": true,
    "emittedReadable": false,
    "encoding": null,
    "endEmitted": false,
    "ended": false,
    "flowing": null,
    "highWaterMark": 16,
    "length": 0,
    "needReadable": true,
    "objectMode": true,
    "paused": true,
    "pipes": null,
    "pipesCount": 0,
    "readableListening": false,
    "reading": false,
    "readingMore": false,
    "resumeScheduled": false,
    "sync": false,
  },
  "_transform": [Function],
  "_transformState": Object {
    "afterTransform": [Function],
    "needTransform": false,
    "transforming": false,
    "writecb": null,
    "writechunk": null,
    "writeencoding": null,
  },
  "_writableState": WritableState {
    "bufferProcessing": false,
    "bufferedRequest": null,
    "bufferedRequestCount": 0,
    "corked": 0,
    "corkedRequestsFree": CorkedRequest {
      "entry": null,
      "finish": [Function],
      "next": null,
    },
    "decodeStrings": true,
    "defaultEncoding": "utf8",
    "destroyed": false,
    "emitClose": true,
    "ended": false,
    "ending": false,
    "errorEmitted": false,
    "finalCalled": false,
    "finished": false,
    "highWaterMark": 16,
    "lastBufferedRequest": null,
    "length": 0,
    "needDrain": false,
    "objectMode": true,
    "onwrite": [Function],
    "pendingcb": 0,
    "prefinished": false,
    "sync": true,
    "writecb": null,
    "writelen": 0,
    "writing": false,
  },
  "allowHalfOpen": true,
  "readable": true,
  "writable": true,
}
`;

exports[`bunyan getSyslogSeverityLevel should properly get a syslog levels 1`] = `
Array [
  "debug",
  "debug",
  "info",
  "warning",
  "err",
  "crit",
]
`;
